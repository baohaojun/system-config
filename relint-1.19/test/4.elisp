;;; Relint test file 4          -*- emacs-lisp -*-

;; Test user-defined function application.
(defun my-char-seq-str (start end)
  (if (<= start end)
      (concat (string start) (my-char-seq-str (1+ start) end))
    ""))

(defun test-fun-call ()
  (looking-at (my-char-seq-str ?$ ?*)))      ; "$%&'()*"

;; Test higher-order functions.
(defun test-hof ()
  (looking-at (apply 'string '(?+ ?x)))
  (looking-at (funcall #'string ?*))
  (looking-at (mapconcat #'identity '("x" "^") "-"))
  (looking-at (if (cl-some 'numberp '('a 2 nil)) "[AA]" ""))
  (looking-at (if (cl-every 'numberp '('a 2 nil)) "" "[BB]"))
  (looking-at (apply 'string (mapcar (lambda (c) (1+ c)) '(?* ?a))))
  (looking-at (apply 'string (mapcan (lambda (c) (list ?+)) '(1 2))))
  (looking-at (apply 'string (sort (list ?$ ?+ ?a) #'<))))

;; Test higher-order functions with keywords.
(defun test-hof-kw ()
  (looking-at (cl-reduce #'concat '(?* ?a ?b) :key #'char-to-string))
  (looking-at (concat "+" (cl-find "c" '("a" "b" "c") :test #'equal))))

;; Test rx
(defvar my-sub-rx '(one-or-more nonl))

;;;(defun test-rx ()    ; Emacs 27-specific test; disable until it's out.
;;;  (looking-at (concat (rx (eval my-sub-rx)
;;;                          (literal (string ?a ?b))
;;;                          (regexp (concat "c" "+"))
;;;                          (regex (string ?d)))
;;;                      "^")))

;; Test macro expansion
(defmacro my-macro (x)
  (list 'string ?a x))

(defun test-macro ()
  (looking-at (my-macro ?^))
  (looking-at (when t "b++"))
  (looking-at (unless nil "c++"))
  (looking-at (string-join `("a" ,@(list "$") ,"b")))
  (looking-at (pcase 'a ((pred symbolp) "d++")))
  (looking-at (cl-case 'z (b "m") (z "*"))))

;; Test repeated use of global variable
(defconst my-var-a "*")
(defconst my-var-b (concat my-var-a "b"))

(defun test-var-ref ()
  (looking-at my-var-b)
  (looking-at (concat my-var-b "c")))

;; Test global variable redefinition
(defconst my-var-a (concat my-var-a "a"))

(defun test-var-redef ()
  (looking-at my-var-a))
