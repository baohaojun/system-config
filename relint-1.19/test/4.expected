4.elisp:10:15: In call to looking-at: Unescaped literal `$' (pos 0)
  "$%&'()*"
   ^
4.elisp:14:15: In call to looking-at: Unescaped literal `+' (pos 0)
  "+x"
   ^
4.elisp:15:15: In call to looking-at: Unescaped literal `*' (pos 0)
  "*"
   ^
4.elisp:16:15: In call to looking-at: Unescaped literal `^' (pos 2)
  "x-^"
   ..^
4.elisp:17:15: In call to looking-at: Duplicated `A' inside character alternative (pos 2)
  "[AA]"
   ..^
4.elisp:18:15: In call to looking-at: Duplicated `B' inside character alternative (pos 2)
  "[BB]"
   ..^
4.elisp:19:15: In call to looking-at: Unescaped literal `+' (pos 0)
  "+b"
   ^
4.elisp:20:15: In call to looking-at: Unescaped literal `+' (pos 0)
  "++"
   ^
4.elisp:21:15: In call to looking-at: Unescaped literal `$' (pos 0)
  "$+a"
   ^
4.elisp:25:15: In call to looking-at: Unescaped literal `*' (pos 0)
  "*ab"
   ^
4.elisp:26:24: In call to looking-at: Unescaped literal `+' (pos 0)
  "+c"
   ^
4.elisp:43:15: In call to looking-at: Unescaped literal `^' (pos 1)
  "a^"
   .^
4.elisp:44:15: In call to looking-at: Repetition of repetition (pos 2)
  "b++"
   ..^
4.elisp:45:15: In call to looking-at: Repetition of repetition (pos 2)
  "c++"
   ..^
4.elisp:46:15: In call to looking-at: Unescaped literal `$' (pos 1)
  "a$b"
   .^
4.elisp:47:15: In call to looking-at: Repetition of repetition (pos 2)
  "d++"
   ..^
4.elisp:48:15: In call to looking-at: Unescaped literal `*' (pos 0)
  "*"
   ^
4.elisp:55:15: In call to looking-at: Unescaped literal `*' (pos 0)
  "*b"
   ^
4.elisp:56:15: In call to looking-at: Unescaped literal `*' (pos 0)
  "*bc"
   ^
4.elisp:62:15: In call to looking-at: Unescaped literal `*' (pos 0)
  "*a"
   ^
